### PR TITLE
feat(workflows): [INFRA-493] Add windows artifact signing workflow

### PR DESCRIPTION
**Pull request by alexs-aero**
- Added windows signing workflow based on Mac signing structure
- Added '--extensions' option that takes multiple extensions for the same type of artifacts
- Added 'win' registry type similar to generic
- Added tests and workflows similar to Mac signing along with test workflows
- Added option to use glob with multiple extensions as well as all
- Included isolation of windows artifacts from NuGet signing.
- New eSigner credentials and ssl.com related account data for signing windows related artifacts.

### CODE REVIEW

File: .github/workflows/deploy-artifacts/tests/bats/test_command_parsers.bats [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+@test "extract_upload_commands finds jf rt upload after ANSI wrapper like dry-run" {
+    local line=$'\033[0;32m   jf rt upload ./app.exe test-project-generic-dev-local/p --flat=false --build-name=b --project=p\033[0m'
+    local cmds
+    cmds=$(extract_upload_commands "$line")
+    echo "$cmds" | grep -qF "jf rt upload ./app.exe"
+    echo "$cmds" | grep -qF "generic-dev-local"
+}
+
+@test "extract_upload_commands finds jf nuget push after ANSI wrapper like dry-run" {
+    local line=$'\033[0;32m   jf nuget push ./pkg.nupkg -Source https://example.local/nuget\033[0m'
```
Comment (albertlagman) [inline .github/workflows/deploy-artifacts/tests/bats/test_command_parsers.bats:16]: I thought extract_upload_commands removes color codes. The color codes just make this harder to read imo. Same for the following test

File: .github/workflows/deploy-artifacts/tests/bats/test_type_registry.bats [hunk 2]
```diff
@@ -37,6 +38,7 @@ setup() {
     [[ "${TYPE_REPO[npm]}" == "npm-dev-local" ]]
     [[ "${TYPE_REPO[pypi]}" == "pypi-dev-local" ]]
     [[ "${TYPE_REPO[go]}" == "go-dev-local" ]]
+    [[ "${TYPE_REPO[win]}" == "generic-dev-local" ]]
```
Comment (albertlagman) [inline .github/workflows/deploy-artifacts/tests/bats/test_type_registry.bats:43]: What's the logic for things like mac, because it doesn't seem to be listed here and i'm pretty sure we already have support for that

File: .github/workflows/deploy-artifacts/tests/bats/test_type_registry.bats [hunk 3]
```diff
@@ -67,20 +70,24 @@ setup() {
...
         [[ "${UPLOAD_ORDER[$i]}" == "pypi" ]] && pypi_idx=$i
         [[ "${UPLOAD_ORDER[$i]}" == "go" ]] && go_idx=$i
+        [[ "${UPLOAD_ORDER[$i]}" == "win" ]] && win_idx=$i
         [[ "${UPLOAD_ORDER[$i]}" == "generic" ]] && generic_idx=$i
     done
     [[ $jar_idx -lt $generic_idx ]]
     [[ $pypi_idx -lt $generic_idx ]]
     [[ $go_idx -lt $generic_idx ]]
+    [[ $win_idx -lt $generic_idx ]]
+    [[ $go_idx -lt $win_idx ]]
```
Comment (albertlagman) [inline .github/workflows/deploy-artifacts/tests/bats/test_type_registry.bats:97]: does go need to be before win? is this some kind of edge case?

**Review by albertlagman — COMMENTED** [review]
Is this ready for submission? It's still marked "draft"

File: .github/workflows/deploy-artifacts/tests/bats/test_command_parsers.bats (hunk 1 above)
Comment (alexs-aero) [reply .github/workflows/deploy-artifacts/tests/bats/test_command_parsers.bats:16]: A good catch. Will update.

**Comment by alexs-aero** [comment]
> Is this ready for submission? It's still marked "draft"

We are waiting for Security team to provide second certificate to sign win artifacts.

File: .github/workflows/deploy-artifacts/tests/bats/test_type_registry.bats (hunk 2 above)
Comment (alexs-aero) [reply .github/workflows/deploy-artifacts/tests/bats/test_type_registry.bats:43]: The trick is that we do not want to have separate folder for win in the end. But as long as we sign them using windows runner, we should place them separately, no NuGet would not find them.

File: .github/workflows/deploy-artifacts/tests/bats/test_type_registry.bats (hunk 3 above)
Comment (alexs-aero) [reply .github/workflows/deploy-artifacts/tests/bats/test_type_registry.bats:97]: Yes, this is for edge cases. Mostly because we are putting win artifacts to generic and we should preserve upload order

File: fakesecrets.env
Comment (albertlagman) [inline fakesecrets.env:1]: I know they're fake but did you mean to include these?

**Review by albertlagman — APPROVED** [review]
Should be okay especially if my comment is not relevant

File: fakesecrets.env
Comment (alexs-aero) [reply fakesecrets.env:1]: This serves two purposes:
- we are checking that workflow processing is doing proper job handling the secrets
- Whoever would need to do this next time, will have an example on how each secret would look like
